src/webui/www/private/confirmdeletion.html

```diff
@@ -1,102 +1,89 @@
+        <span class="deletionGridItem">
+            <button class="disabled" type="button" id="rememberBtn" title="QBT_TR(Remember choice)QBT_TR[CONTEXT=HttpServer]" aria-label="QBT_TR(Remember choice)QBT_TR[CONTEXT=HttpServer]" disabled></button>
+        </span>
+        <span class="deletionGridItem">
+            <input type="checkbox" id="deleteFromDiskCB">
+            <label for="deleteFromDiskCB">
+                <em>QBT_TR(Also remove the content files)QBT_TR[CONTEXT=confirmDeletionDlg]</em>
+            </label>
+        </span>
+    </div>
+</div>
+<div>
+    <input type="button" value="QBT_TR(Remove)QBT_TR[CONTEXT=MainWindow]" id="confirmDeletionButton">
+    <input type="button" value="QBT_TR(Cancel)QBT_TR[CONTEXT=MainWindow]" id="cancelDeletionButton">
+</div>
 
-<head>
-    <meta charset="UTF-8">
-    <title>QBT_TR(Remove torrent(s))QBT_TR[CONTEXT=confirmDeletionDlg]</title>
-    <link rel="stylesheet" href="css/style.css?v=${CACHEID}" type="text/css">
-    <script src="scripts/lib/MooTools-Core-1.6.0-compat-compressed.js"></script>
-    <script src="scripts/lib/MooTools-More-1.6.0-compat-compressed.js"></script>
-    <script>
-        "use strict";
+<script>
+    "use strict";
 
-        function setRememberBtnEnabled(enable) {
-            const btn = $("rememberBtn");
-            btn.disabled = !enable;
+    (() => {
+        const setRememberBtnEnabled = (enable) => {
+            rememberButton.disabled = !enable;
+            rememberButton.classList.toggle("disabled", !enable);
+        };
 
-            const icon = btn.getElementsByTagName("path")[0];
-            if (enable)
-                icon.style.removeProperty("fill");
-            else
-                icon.style.fill = "var(--color-border-default)";
-        }
+        const confirmButton = document.getElementById("confirmDeletionButton");
+        const cancelButton = document.getElementById("cancelDeletionButton");
+        const rememberButton = document.getElementById("rememberBtn");
+        const deleteCB = document.getElementById("deleteFromDiskCB");
+        const deletionText = document.getElementById("deleteTorrentMessage");
 
-        window.addEventListener("DOMContentLoaded", () => {
-            new Request({
-                url: "images/object-locked.svg",
-                method: "get",
-                onSuccess: function(text, xml) {
-                    const newIcon = xml.childNodes[0];
-                    newIcon.style.height = "24px";
-                    newIcon.style.width = "24px";
-                    $("rememberBtn").appendChild(newIcon);
-                    setRememberBtnEnabled(false);
-                }
-            }).send();
+        const {
+            hashes,
+            forceDeleteFiles = false,
+            filterList = null
+        } = window.MUI.Windows.instances["confirmDeletionPage"].options.data;
+        let prefDeleteContentFiles = LocalPreferences.get("delete_torrent_content_files", "false") === "true";
```

Comment:
FYI there was some intertwined logic for deleting the checkbox. Make sure you tested them all:
1. Delete a torrent => checkbox depends on preference setting.
2.  Delete a torrent with shift key => checkbox always checked.

Reply:
Assignment there is for setting a [default value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#default_value) in case when destructured key is not present at all or its value is undefined. Shift + Delete calls deleteFN with forceDeleteFiles set to true. Other windows don't pass forceDeleteFiles at all (just like before) so they fall back on the default value which it then overriden by user preference if needed. Everything should work like before. Should I be more explicit or is it enough to keep it like it is now?

Reply:
If I'm not mistaken. You changed it to read from `LocalPreferences.get("delete_torrent_content_files")` while previously (and correctly) it was read from `prefCache.delete_torrent_content_files`. Those 2 are not the same source.

Reply:
In all my PRs I relied on local storage and I thought moving that setting would be ok in this case as well. Should I change it back?

Reply:
>Should I change it back?

IMO yes.
If some setting is only for webui then it can be stored to `LocalPreferences`. Otherwise it should use `qBittorrent.Cache.preferences`.